### DISCUSSION TITLE
Parcel 2 without package JSON

### DISCUSSION THREAD

**Discussion by jeff-hykin** [parcel-bundler/parcel#7440] [Q&A]
The package.json is a global namespace, and sometimes other tools use it and cause naming conflicts.
Even simple CLI options in parcel 1, like bundling node modules, seem to have no CLI alternative in parcel 2; the global package json is the only option.

Is there a way to use the `--config` arg to get around this? Or a way to specify a different json file + different project root in the CLI?

I looked through the docs, issues, PR, and some of the source code. Couldn't find a conclusive answer, although from what I saw the package.json look very hard coded.

**Comment by devongovett** [comment]
What issues are you running into?

**Reply by jeff-hykin** [reply]
My issue is that build tasks are fragile. Npm init already breaks examples on the website, but directly patching that would just ignore the larger problem of unreliable builds. I have no means of saying
- "hey npm/skypack/vue/electron, I want you have main: blab"
- "hey parcel 2, I want you to have main: stuff"


That is major unpaid technical debt; I can't afford to be under a high pressure deadline, and have both RandomTool123 and Parcel.js 2 decide they are **exclusive** owner of the project-global "targets" name/structure, and the both decided that was the ONLY way they could be configured.


The bigger problem is I need the confidence that I can:
- 1. Add comments to the config options I pick
- 2. Be capable of:
   - not repeating myself (DRY)
    - automating args with the output of another program
   - importing from other sources (package.jsonc, package.yaml, package.toml, .env) 
- 3. Isolate parcel from other processes, including other parcel processes
- All of which are solved by having dynamic rather than static arguments (ENV vars or CLI options)
